### ISSUE TITLE
Armor and Mechanized Infantry units can attack only once per turn

### ISSUE THREAD

**Issue by AlexFolland** [SWY1985/CivOne#300]
I have some Armor and Mechanized Infantry units surrounding an enemy city and each of their turns ended after the first attack.  Their Civilopedia descriptions say "ARMOR units may attack up to three times in one turn." and "MECH. INF. may attack up to three times in one turn." respectively.

**Comment by SWY1985** [comment]
Unit confrontation code is fairly new, so I haven't focussed on unit specifics like how many times a unit may attack. Thanks for this info, it's very useful.

**Comment by AlexFolland** [comment]
I've just tested in Civilization and noticed that Chariots can attack multiple times per turn as well, despite not saying so in in the Civilopedia.  While I haven't tested every unit recently, I think this applies to any land unit with >1 movement points.

**Comment by kazgaroth011** [comment]
Yes, land units can attack as many turns as they have available.

**Comment by SWY1985** [comment]
So, except for the map going out of focus (#280) this issue is resolved?

**Comment by kazgaroth011** [comment]
Yes you can close it.

**Comment by AlexFolland** [comment]
I don't see the commit that fixed this.

**Comment by SWY1985** [comment]
Not solved yet...

**Comment by AlexFolland** [comment]
I will test this more to see if I can reproduce it again.  I was pretty sure of the issue at the time, but I may still have been mistaken if you've seen different behavior.

**Comment by AlexFolland** [comment]
I tested and yes, in this screenshot, I attacked the Legion unit above and to the right of my right-most Chariot unit, killed it, then could not move again with the same Chariot unit that turn.

![image](https://cloud.githubusercontent.com/assets/601517/23830085/de0040ea-06b6-11e7-8503-4bba35e90477.png)

![image](https://cloud.githubusercontent.com/assets/601517/23830100/58bda5e8-06b7-11e7-8d38-65aa806940f0.png)

**Comment by AlexFolland** [comment]
Oddly, this doesn't happen all the time.  I just attacked an enemy city using my Chariot, killed the Phalanx unit inside, then was able to move into the city.

OH!  I think I've figured it out.  It seems to depend on the enemy unit's ground tile.  It is counting forest/swamp/hills/mountain/jungle/tundra/arctic against the attacks, even though I think Civilization doesn't.  I don't remember ever using more than 1 movement point on attacking in Civilization.
